### ISSUE TITLE
Move Localization capabilities to Curseforge

### ISSUE THREAD

**Issue by woopydalan** [DeadlyBossMods/DeadlyBossMods#176]
**Is your feature request related to a problem? Please describe.**
Hi, curseforge has capabilities of using localization in a way that it is easy to maintain and know where translations are lacking. I am using AtlaslootClassic as an example of how this is done.

**Describe the solution you'd like**
Have all the localization strings available on curseforge.

**Describe alternatives you've considered**
Haven't seen a better solution

**Additional context**
Here is the localization table for AtlaslootClassic: 
https://www.curseforge.com/wow/addons/atlaslootclassic/localization

As you can see, every language (minus italian) has 100% translated strings. Therefore, everyone knows what has and has not been translated. If a language was less than 100% translated, it is easy to go into that locale, filter out what needs to be translated, and then do so.

Looking at the Github page of AtlaslootClassic, there is some way to tie in curseforge with Github pages. Here is one example within AtlaslootClassic:
https://github.com/Hoizame/AtlasLootClassic/blob/master/AtlasLootClassic_DungeonsAndRaids/Locales/constants.fr.lua

Somehow, it is pulling the strings from Curseforge and bringing it into Github.

The good news is, no one needs to make a pull request to update translations. Once you enter translations into curseforge, the next time you update DBM on curseforge, it will automatically update all of the translations.

From my perspective, this is very easy to maintain. Currently, when you add a new string, you have to add it to every single locale file (which has not been done). On curseforge, you add one new string, and it adds it to all of the languages. This makes it easier for you as the author. For the translators, it makes it easier because now I don't have to read through every single string in the english locale file and copy it over to my language. Right now it is very cumbersome.

In conclusion, I believe this is the best solution for both you as the author and us as the translators for this project.

Maybe @Hoizame can chime in here?

**Comment by MysticalOS** [comment]
That's def not gonna happen, if they are moved anywhere it'll be a system here somehow. Authors have strongly moved away from curse systems because of how notoriously buggy they are. we automate everything through a 3rd party packager now and only thing curse does is hosts our files after they've been packaged.

**Comment by woopydalan** [comment]
@MysticalOS is there a way to make it transparent which strings are unlocalized? Right now you would have to stay up to date on all changes made and its not feasible for someone to dig through lua files to find missing translations.

**Comment by MysticalOS** [comment]
Eventually maybe, but there are a lot of higher priority things to do by 9.0, including redoing infoframe and rangeframe, as well as updating code to 9.0 standards for setting backdrops, and maybe getting rid of XML and code cleanup. None of that is being done by me either, as I'm already busy with dungeon and raid mods for new expansion. All that stuff I listed is already unpaid volenteer work that progresses very slowly.
